RiotSharp/TournamentEndpoint/TournamentCodeDetail.cs

```diff
@@ -0,0 +1,140 @@
+
+        /// <summary>
+        /// Set of summoner IDs.
+        /// </summary>
+        [JsonProperty("participants")]
+        public HashSet<long> Participants { get; set; }
+
+        /// <summary>
+        /// The password for the tournament code game.
+        /// </summary>
+        [JsonProperty("password")]
+        public string Password { get; set; }
+
+        /// <summary>
+        /// The pick mode for tournament code game.
+        /// </summary>
+        [JsonProperty("pickType")]
+        public TournamentPickType PickType { get; set; }
+
+        /// <summary>
+        /// The provider's ID.
+        /// </summary>
+        [JsonProperty("providerId")]
+        public int ProviderId { get; set; }
+
+        /// <summary>
+        /// The tournament code's region.
+        /// </summary>
+        [JsonProperty("region")]
+        public Region Region { get; set; }
+
+        /// <summary>
+        /// The spectator mode for the tournament code game.
+        /// </summary>
+        [JsonProperty("spectators")]
+        public TournamentSpectatorType Spectators { get; set; }
+
+        /// <summary>
+        /// The team size for the tournament code game.
+        /// </summary>
+        [JsonProperty("teamSize")]
+        public int TeamSize { get; set; }
+
+        /// <summary>
+        /// The tournament's ID.
+        /// </summary>
+        [JsonProperty("tournamentId")]
+        public int TournamentId { get; set; }
+
+        /// <summary>
+        /// Update the pick type, map, spectator type, or allowed summoners for a code.
+        /// </summary>
+        /// <param name="allowedSummonerIds">Comma separated list of summoner Ids.</param>
+        /// <param name="spectatorType">The spectator type.</param>
+        /// <param name="pickType">The pick type.</param>
+        /// <param name="mapType">The map type.</param>
+        public void Update(List<long> allowedSummonerIds, TournamentSpectatorType? spectatorType, TournamentPickType? pickType, TournamentMapType? mapType)
+        {
+            TournamentRiotApi.GetInstance().UpdateTournamentCode(Code, allowedSummonerIds, spectatorType, pickType, mapType);
+        }
+
+        /// <summary>
+        /// Update the pick type, map, spectator type, or allowed summoners for a code asynchronously.
+        /// </summary>
+        /// <param name="allowedSummonerIds">Comma separated list of summoner Ids.</param>
+        /// <param name="spectatorType">The spectator type.</param>
+        /// <param name="pickType">The pick type.</param>
+        /// <param name="mapType">The map type.</param>
+        public async void UpdateAsync(List<long> allowedSummonerIds, TournamentSpectatorType? spectatorType, TournamentPickType? pickType, TournamentMapType? mapType)
+        {
+            await Task.Factory.StartNew(() => TournamentRiotApi.GetInstance().UpdateTournamentCodeAsync(Code, allowedSummonerIds, spectatorType, pickType, mapType));
```

Comment:
As mentioned, if the update method returned `true`/`false` based on the result of the update, you wouldn't have to do `Task.Factory.StartNew`.